src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react/no-danger */
+import { oneLine } from 'common-tags';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import {
+  INCOMPATIBLE_FIREFOX_FOR_IOS,
+  INCOMPATIBLE_NOT_FIREFOX,
+  INCOMPATIBLE_OVER_MAX_VERSION,
+  INCOMPATIBLE_UNDER_MIN_VERSION,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import { sanitizeHTML } from 'core/utils';
+
+import './style.scss';
+
+
+export class AddonCompatibilityErrorBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+    lang: PropTypes.string.isRequired,
+    maxVersion: PropTypes.string.isRequired,
+    minVersion: PropTypes.string.isRequired,
+    reason: PropTypes.string.isRequired,
+    userAgentInfo: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    userAgentInfo: {},
+  }
+
+  render() {
+    const {
+      i18n,
+      lang,
+      maxVersion,
+      minVersion,
+      reason,
+      userAgentInfo,
+    } = this.props;
+    const downloadUrl = `https://www.mozilla.org/${lang}/firefox/`;
+    let message;
+
+    if (typeof maxVersion === 'undefined') {
+      throw new Error('maxVersion is required; it cannot be undefined');
+    }
+    if (typeof minVersion === 'undefined') {
+      throw new Error('minVersion is required; it cannot be undefined');
+    }
+
+    if (reason === INCOMPATIBLE_NOT_FIREFOX) {
+      message = i18n.sprintf(i18n.gettext(oneLine`You need to
+        <a href="%(downloadUrl)s">download Firefox</a> to install this add-on.`
+      ), { downloadUrl });
+    } else if (reason === INCOMPATIBLE_FIREFOX_FOR_IOS) {
+      message = i18n.gettext(
+        'Firefox for iOS does not currently support add-ons.');
+    } else if (reason === INCOMPATIBLE_OVER_MAX_VERSION) {
+      message = i18n.sprintf(i18n.gettext(oneLine`You are using Firefox
+        %(yourVersion)s, but this add-on only supports Firefox up to version
+        %(maxVersion)s.`
+      ), {
+        maxVersion,
+        yourVersion: userAgentInfo.browser.version,
+      });
+    } else if (reason === INCOMPATIBLE_UNDER_MIN_VERSION) {
+      message = i18n.sprintf(i18n.gettext(oneLine`This add-on requires a
+        <a href="%(downloadUrl)s">newer version of Firefox</a> (at least
+        version %(minVersion)s). You are using Firefox %(yourVersion)s.`
+      ), {
+        downloadUrl,
+        minVersion,
+        yourVersion: userAgentInfo.browser.version,
+      });
+    } else {
+      throw new Error('AddonCompatibilityError requires a "reason" prop');
```

Comment:
This isn't the correct place to throw an error. If you get to this `else` branch it means `reason` was undefined (which is better handled up above) *or* we forgot to map an error message to this reason. I think you should allow this latter case and just show a generic message. It would probably be good to log a warning about it though.